ledger/complete/mtrie/forest.go

```diff
@@ -50,7 +50,6 @@ func NewForest(forestCapacity int, metrics module.LedgerMetrics, onTreeEvicted f
 			if !ok {
 				panic(fmt.Sprintf("cache contains item of type %T", value))
 			}
-			//TODO Log error
 			_ = onTreeEvicted(trie)
```

Comment:
in my opinion, `onTreeEvicted` should not throw an error for the following reasons:
* What happens if we receive an error here? What can we do? I think nothing. 
* I think only `onTreeEvicted` has the context to decide what to do. It could log the error itself or crash. Either way, for the forrest there is nothing to do. 

In fact, this is an instance of the anti-pattern of "continue on a best effort basis, when we receive an error". It is very important to me that we avoid this anti-pattern, because it is fundamentally incompatible with BFT. Would you be able to clean this up (i.e. remove the error return) as part of this PR? That would be great. Thanks.

Reply:
> Would you be able to clean this up (i.e. remove the error return) as part of this PR? That would be great. Thanks.

[Done](https://github.com/onflow/flow-go/pull/1944/commits/c99bc8c9128a56bc0569c38f91964432c7cbe4d3) (cleaned up and improved old error logging as part of this PR).  It was going to be part of issue #2074.

Great suggestion about removing the returned error.  👍  Thanks for detailed and thoughtful explanation.  💎